src/qibosoq/programs/base.py

```diff
@@ -274,9 +275,11 @@ def collect_shots(self) -> Tuple[npt.NDArray[np.float64], npt.NDArray[np.float64
                 lengths.append(self.soc.us2cycles(pulse.duration, gen_ch=ro_ch))
             adcs.append(adc_ch)
 
-        adcs, adc_count = np.unique(adcs, return_counts=True)
+        unique_adcs, adc_count = np.unique(adcs, return_counts=True)
 
-        for idx, adc_ch in enumerate(adcs):
+        shape: Tuple  # used for mypy
```

Comment:
Convert below in an `if` expression:
```py
shape: Tuple = (count, self.expts, self.reps) if self.expts is not None else (count, self.reps)
```

Also consider that tuples are like structs with anonymous fields, so having the same variable with two different lengths is like making a union, even though the union is just named `Tuple` (that here could also be `tuple`).
However, NumPy accepts the union, so it is consistent. This is just a warning, not a change request.